src/control/common/file_utils.go

```diff
@@ -219,3 +221,22 @@ func SyncDir(path string) (err error) {
 
 	return d.Sync()
 }
+
+// CheckSudo verifies process is running as root.
+func CheckSudo() (ok bool, err error) {
+	cmd := exec.Command("id", "-u")
+	output, err := cmd.Output()
+	if err != nil {
+		return
+	}
+
+	// trim trailing new-line char
+	// 0 = root, 501 = non-root user
```

Comment:
Isn't it any non-zero user isn't root? Also there is an easier way to do this in go. the user package has a call called current which will give you a user struct with this information filled out. I don't know if the UID in that struct is the euid or not (also its a string not an int). If it isn't then you can try os.Geteuid which will return you the int of the effective uid.

https://golang.org/pkg/os/#Geteuid

Reply:
are you requesting a change with regard to this?